langchain/embeddings/huggingface.py

```diff
@@ -53,18 +74,34 @@ def embed_documents(self, texts: List[str]) -> List[List[float]]:
             List of embeddings, one for each text.
         """
         texts = list(map(lambda x: x.replace("\n", " "), texts))
-        embeddings = self.client.encode(texts)
+
+        if (self.model_type == MODEL_TYPE.INSTRUCTION_EMBEDDING):
+            instruction_pairs = []
+            for text in texts:
+                instruction_pairs.append([self.instruction, text])
+            embeddings = self.client.encode(instruction_pairs)
+        else:
+            embeddings = self.client.encode(texts)
+
+        if (self.model_name == DEFAULT_MODEL_NAME):
+            return embeddings.tolist()
+
         return embeddings.tolist()
 
     def embed_query(self, text: str) -> List[float]:
         """Compute query embeddings using a HuggingFace transformer model.
-
         Args:
             text: The text to embed.
 
         Returns:
             Embeddings for the text.
         """
         text = text.replace("\n", " ")
-        embedding = self.client.encode(text)
+
+        if (self.model_type == MODEL_TYPE.INSTRUCTION_EMBEDDING):
+            instruction_pair = [self.instruction, text]
+            embedding = self.client.encode(instruction_pair)
```

Comment:
Following the example on the Instructor repo page I had to put the instruction pair inside a list and then extract just the first element of the response in order to get the format matching the sentence transformer version as per my comment in the test. Something like this:
```suggestion
            embedding = self.client.encode([instruction_pair])[0]
```

Reply:
Ah I see, because a list of tensors (Len==1) is returned, we need to grab first. Good catch.